vdr/didnuts/manager.go

```diff
@@ -67,3 +95,193 @@ func (m Manager) UpdateService(_ context.Context, _ did.DID, _ ssi.URI, _ did.Se
 func (m Manager) DeleteService(_ context.Context, _ did.DID, _ ssi.URI) error {
 	return fmt.Errorf("DeleteService() is not supported for did:%s", MethodName)
 }
+
+func (m Manager) NewDocument(ctx context.Context, keyFlags didsubject.DIDKeyFlags) (*didsubject.DIDDocument, error) {
+	// First, generate a new keyPair with the correct kid
+	// Currently, always keep the key in the keystore. This allows us to change the transaction format and regenerate transactions at a later moment.
+	// Relevant issue:
+	// https://github.com/nuts-foundation/nuts-node/issues/1947
+	key, err := m.KeyStore.New(ctx, didKIDNamingFunc)
+	if err != nil {
+		return nil, err
+	}
+
+	keyID, err := did.ParseDIDURL(key.KID())
+	if err != nil {
+		return nil, err
+	}
+
+	didID, _ := resolver.GetDIDFromURL(key.KID())
+	var verificationMethod *did.VerificationMethod
+	// Add VerificationMethod using generated key
+	verificationMethod, err = did.NewVerificationMethod(*keyID, ssi.JsonWebKey2020, didID, key.Public())
+	if err != nil {
+		return nil, err
+	}
+	vmAsJson, _ := json.Marshal(verificationMethod)
+	now := time.Now().Unix()
+	sqlDoc := didsubject.DIDDocument{
+		DID: didsubject.DID{
+			ID: didID.String(),
+		},
+		CreatedAt: now,
+		UpdatedAt: now,
+		VerificationMethods: []didsubject.VerificationMethod{
+			{
+				ID:       verificationMethod.ID.String(),
+				KeyTypes: didsubject.VerificationMethodKeyType(keyFlags),
+				Data:     vmAsJson,
+			},
+		},
+	}
+
+	return &sqlDoc, nil
+}
+
+func (m Manager) NewVerificationMethod(ctx context.Context, controller did.DID, _ didsubject.DIDKeyFlags) (*did.VerificationMethod, error) {
+	// did:nuts uses EC keys for everything so it doesn't use the DIDKeyFlags
+	return CreateNewVerificationMethodForDID(ctx, controller, m.KeyStore)
```

Comment:
all dids are now self-controlled, so the term `controller` will become weirder over time. It's used several times in this file. Not sure if its worth changing though